subsys/logging/log_output_syst.c

```diff
@@ -684,6 +685,85 @@ void log_output_hexdump_syst_process(const struct log_output *log_output,
 	MIPI_SYST_WRITE(&log_syst_handle, severity, 0x1A, data, length);
 }
 
+#else
+static void hexdump2_print(struct log_msg2 *msg,
+			  const struct log_output *output)
+{
+	char *buf;
+	size_t length;
+	uint32_t severity = level_to_syst_severity(log_msg2_get_level(msg));
+
+	buf = log_msg2_get_data(msg, &length);
+	MIPI_SYST_WRITE(&log_syst_handle, severity, 0x1A, buf, length);
+}
+
+void log_output_msg2_syst_process(const struct log_output *output,
+				struct log_msg2 *msg, uint32_t flag)
+{
+	uint32_t severity = level_to_syst_severity(log_msg2_get_level(msg));
+	size_t len, hexdump_len;
+
+	update_systh_platform_data(&log_syst_handle, output, flag);
+
+	uint8_t *data = log_msg2_get_package(msg, &len);
+
+	log_msg2_get_data(msg, &hexdump_len);
+
+	if (len && !hexdump_len) {
+
+		char *buf = data, *fmt;
+		unsigned int args_size;
+
+		fmt = ((char **)buf)[1];
+		args_size = ((((uint8_t *)buf)[0] * sizeof(int))/4)-2;
+		buf += sizeof(char *) * 2;
+
```

Comment:
Do you know if it would be possible to extend mipi syst library and add `MIPI_SYS_VPRINTF(h, sev, fmt, ap)`? It would be rather trivial change but i don't how library is supported.

That would help here a lot as after getting `ap` and `fmt` from cbprintf package you would directly call `MIPI_SYS_VPRINTF`.

Current solution has severe limitations, up to 4 args and only `int` args.

Reply:
@nordic-krch That is a very good point. MIPI sys-t doesn't support any VPRINTF type API AFAIK. But have added upto 15 switch cases to extend the support. It can only support int args for now (Note: v1 logging in general and with sys-t as well also doesn't support float values). I can add the support for other parameter types as a followup improvement to this. Let me know your thoughts.

Reply:
what about adding vprintf support to mipi sys-t. I see that we are maintaining a zephyr fork so it would be straightforward to add it there. Don't know who maintains upstream project but since it is not very controversial change i guess that we could upstream it as well. That would solve all the issue. @nashif do you know if upstream project is opened for patches like that?

Reply:
will check

Reply:
@nordic-krch have the VPRINTF supported now [here](https://github.com/zephyrproject-rtos/zephyr/pull/41277). This will also support floating point variables alongwith printing variable number of arguments. Please re-review and let me know if any other changes are needed.